File: primer-curso/programacion/unidad-02-Java&Python/python/GraficaNumeros.py
numeros = [-6,2,5,-14,15,3,12,5,-4,7,1]
menor = 5
texto =""

print(numeros)
menor = abs(min(numeros))


for i in numeros:
	if i >=0:
		espacios = (menor) 
		texto += " " * espacios
		texto+="|"
		texto+="*" * i

	if i < 0:
		espacios = (menor) - abs(i)
		texto += " " * espacios
		texto+= "*" * abs(i)
		texto+= "|"

	texto+="\n"

print(texto)

File: primer-curso/programacion/unidad-02-Java&Python/java/GraficaNumeros.java
public class GraficaNumeros{
	public static void main(String[] args){

		int[] numeros = {5,24,-1,6,-13,-5,-10,21};
		int menor = 100000000;

		//Asignamos el menor numero a la variable
		for(int i = 0; i < numeros.length; i++){
			if (numeros[i] < menor){
				menor = numeros[i];
			}
		}

		//Recorremos el array para imprimir los caracteres
		for(int i = 0; i < numeros.length;i++){
			//Si el numero es negativo o menor que 0 ejecutamos lo siguiente
			if (numeros[i] < 0){
				//Printeamos la primera parte de espacios
				for(int j = 0; j < (Math.abs(menor) - Math.abs(numeros[i]));j++){
					System.out.print(" ");
				}
				for(int j = 0; j < Math.abs(numeros[i]);j++){
					System.out.print("*");
				}
				System.out.print("|\n");
			}
			else if(numeros[i] >= 0){
				for(int j = 0; j < Math.abs(menor);j++){
					System.out.print(" ");
				}

				System.out.print("|");

				for(int j = 0; j < Math.abs(numeros[i]);j++){
					System.out.print("*");
				}
				System.out.print("\n");
			}

		}

	}
}